tests/sqlparser_snowflake.rs

```diff
@@ -2952,3 +2950,19 @@ fn test_sf_double_dot_notation() {
 
 #[test]
 fn test_parse_double_dot_notation_wrong_position() {}
+
+#[test]
+fn test_table_sample() {
+    snowflake_and_generic().one_statement_parses_to(
+        "SELECT * FROM testtable SAMPLE (10)",
+        "SELECT * FROM testtable TABLESAMPLE (10)",
+    );
```

Comment:
Oh could we do the same here as well? clickhouse for example doesn't have has `SAMPLE` but not `TABLESAMPLE` so that the roundtrip stays correct

Reply:
Added support for clickhouse as well, with a different approach to how to model the it in the AST